File: gcj_2014/QR_D.py
#!/usr/bin/env python

# Google Code Jam 2014 Qualification Round
# Problem D. Deceitful War

import sys

def deceitful(a, b):
  a.sort()
  b.sort()
  s = 0
  while len(a) > 1:
    if a[0] < b[0]:
      a.pop(0)
      b.pop()
    else:
      s += 1
      a.pop(0)
      b.pop(0)
  if a[0] > b[0]:
    s += 1
  return s

def war(a, b):
  a.sort()
  b.sort()
  s = 0
  while len(a) > 0:
    x = a[0]
    a.pop(0)
    for y in b:
      if y > x:
        break
    if x > y:
      s += 1
      b.pop(0)
    else:
      b.remove(y)
  return s

def solve(a, b):
  return [deceitful(a[:], b[:]), war(a, b)]

T = int(sys.stdin.readline())
for t in range(1, T+1):
  N = int(sys.stdin.readline())
  a = [float(n) for n in sys.stdin.readline().rstrip("\n").split(" ")]
  b = [float(n) for n in sys.stdin.readline().rstrip("\n").split(" ")]
  ans = solve(a, b)
  print("Case #%d: %d %d" % (t, ans[0], ans[1]))

File: gcj_2016/QR_D.go
// Google Code Jam 2016 Qualification Round
// Problem D. Fractiles

package main

import (
    "bufio"
    "fmt"
    "os"
    "strconv"
)

var sc = bufio.NewScanner(os.Stdin)

func nextInt() int {
    sc.Scan()
    i, e := strconv.Atoi(sc.Text())
    if e != nil {
        panic(e)
    }
    return i
}

func main() {
    sc.Split(bufio.ScanWords)
    T := nextInt()
    for t := 1; t <= T; t++ {
	k := nextInt()
	c := nextInt()
	s := nextInt()
	fmt.Print("Case #")
	fmt.Print(t)
	fmt.Print(":")
	for i := 1; i <= k; i++ {
	    fmt.Print(" ")
	    fmt.Print(i)
        }
	fmt.Println("")
	_ = c
	_ = s
    }
}
